### PR TITLE
Tests: test that no worlds fail to load

### PR DESCRIPTION
**Pull request by Silvris**
## What is this fixing or adding?
Adds a test that looks at the current status of worlds.failed_world_loads and fails if any world failed to load. Worlds that fail to load will not have any other tests ran for them, so this should catch any major instance of failing.

## How was this tested?
Ran unittests, confirmed currently failing on python 3.8.

### CODE REVIEW

File: test/general/test_implemented.py [hunk 1]
```diff
@@ -47,3 +48,8 @@ def test_slot_data(self):
                 for key, data in multiworld.worlds[1].fill_slot_data().items():
                     self.assertIsInstance(key, str, "keys in slot data must be a string")
                     self.assertIsInstance(encode(data), str, f"object {type(data).__name__} not serializable.")
+
+    def test_no_failed_world_loads(self):
+        if failed_world_loads:
+            self.fail(f"The following worlds failed to load: {failed_world_loads}")
+
```
Comment (black-sliver) [inline test/general/test_implemented.py:55]: extra newline at end of file

**Review by black-sliver — COMMENTED** [review]
It's somewhat amazing that we never realized this problem existed. I think having *any* test in the world itself would catch that :S

Have you pinged the world authors already?

File: test/general/test_implemented.py (hunk 1 above)
Comment (Silvris) [reply test/general/test_implemented.py:55]: There's only supposed to be one newline at the end of the file right? There's only one. I think initially pushing without it seems to confuse Github for some reason.

**Comment by Silvris** [comment]
MLSS is known and addressed in #3286 .

Yugioh is new and is what inspired this test to be made.

File: test/general/test_implemented.py (hunk 1 above)
Comment (Silvris) [reply test/general/test_implemented.py]: Nevermind, Github Desktop is what got confused.

**Review by NewSoupVi — APPROVED** [review]

**Review by beauxq — APPROVED** [review]
I didn't test it myself, but I saw it fail in this PR before the 2 worlds were fixed.

**Review by black-sliver — APPROVED** [review]

**Pull request merged**